crates/ide-completion/src/context/analysis.rs

```diff
@@ -359,7 +359,8 @@ fn expected_type_and_name(
                     let ty = it.pat()
                         .and_then(|pat| sema.type_of_pat(&pat))
                         .or_else(|| it.initializer().and_then(|it| sema.type_of_expr(&it)))
-                        .map(TypeInfo::original);
+                        .map(TypeInfo::original)
+                        .filter(|ty| it.ty().is_some() || !ty.is_fn());
```

Comment:
What I'm doing here is only taking `Fn` types into account for the let statement if it is explicitly defined. Is there a case I'm not seeing where this will be problematic?

Reply:
Don't think so but please add a commend for why this is done.